java/client/src/org/openqa/selenium/edge/EdgeDriver.java

```diff
@@ -18,139 +18,125 @@
- *     driver.get("http://www.google.com");
- *     WebElement searchBox = driver.findElement(By.name("q"));
- *     searchBox.sendKeys("webdriver");
- *     searchBox.quit();
- *     assertEquals("webdriver - Google Search", driver.getTitle());
- *   }
- * }
- * }</pre>
+ *     {@Literal @BeforeEach}
+ *     public void createDriver() {
+ *         driver = new RemoteWebDriver(service.getUrl(),
+ *                 new EdgeOptions());
+ *     }
  *
+ *     {@Literal @AfterEach}
+ *     public void quitDriver() {
+ *         driver.quit();
+ *     }
  *
- * @see EdgeDriverService#createDefaultService
+ *     {@Literal @Test}
+ *     public void testBingSearch() {
+ *         driver.get("http://www.bing.com");
+ *         WebElement searchBox = driver.findElement(By.name("q"));
+ *         searchBox.sendKeys("webdriver");
+ *         searchBox.submit();
+ *         assertEquals("webdriver - Bing", driver.getTitle());
+ *     }
+ * }}</pre>
  */
-public class EdgeDriver extends RemoteWebDriver {
+public class EdgeDriver extends ChromeDriver {
+
+  /**
+   * Boolean system property that defines whether the msedgedriver executable (Chromium Edge)
+   * should be used.
+   */
+  public static final String DRIVER_USE_EDGE_CHROMIUM = "webdriver.edge.chromium";
+
+  public EdgeDriver() { this(new EdgeOptions()); }
+
+  public EdgeDriver(EdgeOptions options) {
+    super(toExecutor(options), options);
+  }
 
-	  /**
-	   * Creates a new EdgeDriver using the {@link EdgeDriverService#createDefaultService default}
-	   * server configuration.
-	   *
-	   * @see #EdgeDriver(EdgeDriverService, EdgeOptions)
-	   */
-	  public EdgeDriver() {
-		    this(EdgeDriverService.createDefaultService(), new EdgeOptions());
-		  }
+  @Deprecated
+  public EdgeDriver(Capabilities capabilities) {
+    super(toExecutor(new EdgeOptions()), capabilities);
+  }
 
-	  /**
-	   * Creates a new EdgeDriver instance. The {@code service} will be started along with the driver,
-	   * and shutdown upon calling {@link #quit()}.
-	   *
-	   * @param service The service to use.
-	   * @see #EdgeDriver(EdgeDriverService, EdgeOptions)
-	   */
-	  public EdgeDriver(EdgeDriverService service) {
-	    this(service, new EdgeOptions());
-	  }
+  private static CommandExecutor toExecutor(EdgeOptions options) {
+    Objects.requireNonNull(options, "No options to construct executor from");
 
-	  /**
-	   * Creates a new EdgeDriver instance. The {@code capabilities} will be passed to the
-	   * EdgeDriver service.
-	   *
-	   * @param capabilities The capabilities required from the EdgeDriver.
-	   * @see #EdgeDriver(EdgeDriverService, Capabilities)
-     * @deprecated Use {@link EdgeDriver(EdgeOptions)}
-	   */
-	  @Deprecated
-	  public EdgeDriver(Capabilities capabilities) {
-	    this(EdgeDriverService.createDefaultService(), capabilities);
-	  }
+    EdgeDriverService.Builder<?, ?> builder =
+        StreamSupport.stream(ServiceLoader.load(DriverService.Builder.class).spliterator(), false)
```

Comment:
I tried to use if/else block, but edge module can't find EdgeHTMLDriverService code since it's not built yet, but I can't have edge has a dependency on edgehtml module. It's be a circular dependency because edgehtml module already has edge as its dependency. That's why I figured using FirefoxDriver pattern to get it working :)